src/sentry/static/sentry/app/views/events/eventsChart.jsx

```diff
@@ -162,6 +164,52 @@ class EventsChart extends React.Component {
       },
     };
 
+    let chartImplementation = ({
+      zoomRenderProps,
+      releaseSeries,
+      errored,
+      loading,
+      reloading,
+      timeseriesData,
+      previousTimeseriesData,
+    }) => {
+      if (errored) {
+        return (
+          <ErrorPanel>
+            <IconWarning color={theme.gray2} size="lg" />
+          </ErrorPanel>
+        );
+      }
+
+      return (
+        <TransitionChart loading={loading} reloading={reloading}>
+          <TransparentLoadingMask visible={reloading} />
+          <EventsAreaChart
+            {...zoomRenderProps}
+            tooltip={tooltip}
+            loading={loading}
+            reloading={reloading}
+            utc={utc}
+            showLegend={showLegend}
+            releaseSeries={releaseSeries || []}
+            timeseriesData={timeseriesData}
+            previousTimeseriesData={previousTimeseriesData}
+            currentSeriesName={currentSeriesName}
+            previousSeriesName={previousSeriesName}
+          />
+        </TransitionChart>
+      );
+    };
+
+    if (!disableReleases) {
+      const previousChart = chartImplementation;
+      chartImplementation = chartProps => (
```

Comment:
I don't love this pattern but couldn't come up with something better that wasn't also more complicated/indirect.